core/src/main/java/org/apache/calcite/sql/SqlOverOperator.java

```diff
@@ -78,6 +78,12 @@ public SqlOverOperator() {
       throw validator.newValidationError(aggCall, RESOURCE.overNonAggregate());
     }
     final SqlNode window = call.operand(1);
+    SqlLiteral qualifier = aggCall.getFunctionQuantifier();
```

Comment:
Can we move this validation to a separate method? See this for reference: org.apache.calcite.sql.SqlFunction#validateQuantifier

Reply:
Done